ivadomed/loader/generalized_loader_configuration.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
```

Comment:
I have some questions about this file (which I don't think we covered during the meeting).

Is it only used for the V2 loader?
Does it supersede the default configuration keys from `config_default.json` that are set by `config_manager.py` when running ivdomed?

Reply:
Ok so ultimetately, will we need to maintain both config_default.json and this file up-to-date or only this file?

Reply:
This file will hopefully replace all the above signatures parameters at the end of BIDSDataframe update. FileDataset no longer use above parameters anymore. It uses this class instead (same information, organized slightly differently)

BIDSDataframe will be udpated to use it during its upcoming rewrite. 

MRI2DSegmentation Dataset constructor (shown above) will also by then use the same configuration object. 

User perspective, they will be using V2 standard. User do not touch this configuration class file. This class is used to internally store and represents all the configurations to simplify interfaces and provide internal validations/consistency checks. Nothing much more than a a bit fancier tuple structure in a sense.